psydac/linalg/stencil.py

```diff
@@ -1629,77 +1601,41 @@ def _prepare_transpose_args(self):
         cm    = W.shifts
 
         # Number of rows in the transposed matrix (along each dimension)
-        nrows       = [e-s+1 for s,e in zip(ssd, eed)]
-        ncols       = [e-s+2*m*p+1 for s,e,m,p in zip(ssc, eec, cm, gpads)]
+        nrows       = [e-s+1 for s, e in zip(ssd, eed)]
+        ncols       = [e-s+2*m*p+1 for s, e, m, p in zip(ssc, eec, cm, gpads)]
 
         pp = pads
-        ndiags, starts = list(zip(*[compute_diag_len(p,mi,mj, return_padding=True) for p,mi,mj in zip(pp,cm,dm)]))
-        ndiagsT, _     = list(zip(*[compute_diag_len(p,mj,mi, return_padding=True) for p,mi,mj in zip(pp,cm,dm)]))
+        ndiags, starts = list(zip(*[compute_diag_len(p, mi, mj, return_padding=True) for p, mi, mj in zip(pp, cm, dm)]))
+        ndiagsT, _     = list(zip(*[compute_diag_len(p, mj, mi, return_padding=True) for p, mi, mj in zip(pp, cm, dm)]))
 
-        diff   = [gp-p for gp,p in zip(gpads, pp)]
+        diff   = [gp-p for gp, p in zip(gpads, pp)]
 
-        sl   = [(s if mi>mj else 0) + (s%mi+mi//mj if mi<mj else 0)+(s if mi==mj else 0)\
-                 for s,p,mi,mj in zip(starts,pp,cm,dm)]
+        sl   = [(s if mi > mj else 0) + (s % mi + mi//mj if mi < mj else 0)+(s if mi == mj else 0)\
+                 for s, p, mi, mj in zip(starts, pp, cm, dm)]
 
-        si   = [(mi*p-mi*(int(np.ceil((p+1)/mj))-1) if mi>mj else 0)+\
-                 (mi*p-mi*(p//mi)+ d*(mi-1) if mi<mj else 0)+\
-                 (mj*p-mj*(p//mi)+ d*(mi-1) if mi==mj else 0)\
-                  for mi,mj,p,d in zip(cm, dm, pp, diff)]
+        si   = [(mi * p - mi * (int(np.ceil((p + 1)/mj)) - 1) if mi > mj else 0) + \
+                 (mi * p - mi * (p//mi) + d * (mi - 1) if mi < mj else 0) + \
+                 (mj * p - mj * (p//mi) + d * (mi - 1) if mi == mj else 0)\
+                  for mi, mj, p, d in zip(cm, dm, pp, diff)]
 
         sk   = [n-1\
-                 + (-(p%mj) if mi>mj else 0)\
-                 + (-p+mj*(p//mi) if mi<mj  else 0)\
-                 + (-p+mj*(p//mi) if mi==mj else 0)\
-                 for mi,mj,n,p in zip(cm, dm, ndiagsT, pp)]
-
-        args = {}
-        args['nrows']   = tuple(nrows)
-        args['ncols']   = tuple(ncols)
-        args['gpads']   = tuple(gpads)
-        args['pads']    = tuple(pads)
-        args['dm']      = tuple(dm)
-        args['cm']      = tuple(cm)
-        args['ndiags']  = tuple(ndiags)
-        args['ndiagsT'] = tuple(ndiagsT)
-        args['si']      = tuple(si)
-        args['sk']      = tuple(sk)
-        args['sl']      = tuple(sl)
-        return args
+                 + (-(p % mj) if mi > mj else 0)\
+                 + (-p + mj * (p//mi) if mi < mj else 0)\
+                 + (-p + mj * (p//mi) if mi == mj else 0)\
+                 for mi, mj, n, p in zip(cm, dm, ndiagsT, pp)]
+
+        args = [*nrows, *ncols, *gpads, *pp, *dm, *cm, *ndiags, *ndiagsT, *si, *sk, *sl]
+        return np.int64(args)
```

Comment:
Originally `args` was a dictionary, so one could pass it as `**args` without worrying about ordering the arguments. In the current solution `args` is a list, so the order of the arguments becomes very important. Are you sure that this is a robust choice?